docs/ASDK-2.0-Layout-Documention.md

```diff
@@ -0,0 +1,802 @@
+    * [@property (nonatomic, assign) ASStackLayoutAlignItems alignItems;](#property-nonatomic-assign-asstacklayoutalignitems-alignitems)
+    * [@property (nonatomic, assign) BOOL baselineRelativeArrangement;](#property-nonatomic-assign-bool-baselinerelativearrangement)
+  * [The ASStackLayoutable protocol](#the-asstacklayoutable-protocol)
+    * [@property (nonatomic, readwrite) CGFloat spacingBefore;](#property-nonatomic-readwrite-cgfloat-spacingbefore)
+    * [@property (nonatomic, readwrite) CGFloat spacingAfter;](#property-nonatomic-readwrite-cgfloat-spacingafter)
+    * [@property (nonatomic, readwrite) BOOL flexGrow;](#property-nonatomic-readwrite-bool-flexgrow)
+    * [@property (nonatomic, readwrite) BOOL flexShrink;](#property-nonatomic-readwrite-bool-flexshrink)
+    * [@property (nonatomic, readwrite) ASDimension flexBasis;](#property-nonatomic-readwrite-asdimension-flexbasis)
+    * [@property (nonatomic, readwrite) ASStackLayoutAlignSelf alignSelf;](#property-nonatomic-readwrite-asstacklayoutalignself-alignself)
+    * [@property (nonatomic, readwrite) CGFloat ascender;](#property-nonatomic-readwrite-cgfloat-ascender)
+    * [@property (nonatomic, readwrite) CGFloat descender;](#property-nonatomic-readwrite-cgfloat-descender)
+  * [The ASStaticLayoutable protocol](#the-asstaticlayoutable-protocol)
+    * [@property (nonatomic, assign) CGPoint layoutPosition;](#property-nonatomic-assign-cgpoint-layoutposition)
+* [Layout Specs](#layout-specs)
+* [Implementing layoutSpecThatFits:](#implementing-layoutspecthatfits)
+* [Types of layout specs](#types-of-layout-specs)
+  * [ASLayoutSpec](#aslayoutspec)
+  * [ASWrapperLayoutSpec](#aswrapperlayoutspec)
+  * [ASInsetLayoutSpec](#asinsetlayoutspec)
+  * [ASOverlayLayoutSpec](#asoverlaylayoutspec)
+  * [ASBackgroundLayoutSpec](#asbackgroundlayoutspec)
+  * [ASCenterLayoutSpec](#ascenterlayoutspec)
+  * [ASRatioLayoutSpec](#asratiolayoutspec)
+  * [ASRelativeLayoutSpec](#asrelativelayoutspec)
+  * [ASStackLayoutSpec: Flexbox Container](#asstacklayoutspec-flexbox-container)
+  * [ASStaticLayoutSpec: Absolute Container](#asstaticlayoutspec-absolute-container)
+* [Important methods](#important-methods)
+  * [-[ASLayoutable layoutThatFits:]](#-aslayoutable-layoutthatfits)
+  * [-[ASLayoutable calculateLayoutThatFits:]](#-aslayoutable-calculatelayoutthatfits)
+  * [-[ASLayoutable calculateLayoutThatFits:restrictedToSize:relativeToParentSize:]](#-aslayoutable-calculatelayoutthatfitsrestrictedtosizerelativetoparentsize)
+* [Determine the best size for a layoutable](#determine-the-best-size-for-a-layoutable)
+* [Manual Layout](#manual-layout)
+  * [Manual Layout UIKit](#manual-layout-uikit)
+  * [Manual Layout ASDK](#manual-layout-asdk)
+    * [calculateSizeThatFits and <code>layout</code>](#calculatesizethatfits-and-layout)
+    * [Example](#example)
+* [Ressources:](#ressources)
+
+## Introduction
+// TODO: Small introduction with an example
+
+## Differences from web
```

Comment:
This section spend a lot of time explaining what the layout system _isn't_, but not a lot of what it really is. Best to introduce newcomers on the strengths, benefits and positive elements of the framework before directly comparing it to something else like the CSS Box Model. Here is really the time to sell and convince them of how simple and understandable the whole system is.